tests/examples/heroku_deploy/script.protocols.yaml

```diff
@@ -0,0 +1,45 @@
+protocols:
+  # exits when the git repo is not clean
+  - env: &env
+      PATH: /snap/bin:~/.cargo/bin:/usr/bin:/bin
+    arguments: test-heroku-app
+    protocol:
+      - command: git status --porcelain
+        stdout: " M some-changed-file\n"
+    stderr: "exiting, git repo not clean\n"
+    exitcode: 1
+
+  # exits and prints a usage message when no arguments given
+  - env: *env
```

Comment:
@matthandlersux: Look at those fancy yaml features that just work. :)

Reply:
it's pretty nice that we get that for free with the yaml parsing library